### PR TITLE
feat(iam/policy): support NotAction field

### PR DESCRIPTION
**Pull request by eirture**
I'm trying to fix minio/minio#15328 to implement `NotAction` field in a policy statement.

### CODE REVIEW

**Comment by eirture** [comment]
Hi @harshavardhana , I'm trying to implement NotAction in a policy statement. PTAL thx! 🙏

**Review by krisis — APPROVED** [review]

File: iam/policy/statement.go [hunk 1]
```diff
@@ -166,8 +171,14 @@ func (statement Statement) Equals(st Statement) bool {
...
-	return NewStatement(statement.SID, statement.Effect, statement.Actions.Clone(),
-		statement.Resources.Clone(), statement.Conditions.Clone())
+	return Statement{
+		SID:        statement.SID,
+		Effect:     statement.Effect,
+		Actions:    statement.Actions,
+		NotActions: statement.NotActions,
+		Resources:  statement.Resources,
+		Conditions: statement.Conditions,
+	}
```
Comment (harshavardhana) [inline iam/policy/statement.go:181]: why is this being changed? @eirture

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — CHANGES_REQUESTED** [review]
We need more tests here to cater for follow styles

- "Allow" with NotAction 
- "Deny" with NotAction

https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_notaction.html

File: iam/policy/statement.go (hunk 1 above)
Comment (eirture) [reply iam/policy/statement.go:181]: This method is a clone operation. A new field `NoActions` add to the statement, when we clone a statement we should copy the `NotActions` too. Is this right?

**Comment by eirture** [comment]
> We need more tests here to cater for follow styles
> 
> * "Allow" with NotAction
> * "Deny" with NotAction
> 
> https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_notaction.html

@harshavardhana  thx for review. I will add more tests for this case.

File: iam/policy/statement.go (hunk 1 above)
Comment (harshavardhana) [reply iam/policy/statement.go:181]: yes but you need to enhance the NewStatement() to take `NotActions.Clone()` as input what is happening right now is not a copy AFAIK.

Comment (eirture) [reply iam/policy/statement.go:181]: yes but I think the NewStatement function is a promise to users, we can't change parameters anymore. Can we add a new function to create a statement? Could you give me some advice?

Comment (harshavardhana) [reply iam/policy/statement.go:181]: yes you can of course add, the only consumer of this function is MinIO - we don't expect users to use this package outside of MinIO. That is out scope for us.

Comment (harshavardhana) [reply iam/policy/statement.go:181]: you can also add NewStatementWithNotActions() if you want to keep previous function as is.

**Comment by eirture** [comment]
Hi @harshavardhana @krisis, I have pushed some updates about review comments. PTAL again, thx! 🍺

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**